inference-engine/tests/functional/inference_engine/ir_serialization/serialize.cpp

```diff
@@ -48,18 +73,32 @@ TEST_P(SerializationTest, CompareFunctions) {
 }
 
 INSTANTIATE_TEST_CASE_P(IRSerialization, SerializationTest,
-        testing::Values(std::make_tuple("add_abc.xml"),
-                        std::make_tuple("add_abc_f64.xml"),
-                        std::make_tuple("split_equal_parts_2d.xml"),
-                        std::make_tuple("addmul_abc.xml"),
-                        std::make_tuple("add_abc_initializers.xml"),
-                        std::make_tuple("experimental_detectron_roi_feature_extractor.xml"),
-                        std::make_tuple("experimental_detectron_detection_output.xml"),
-                        std::make_tuple("nms5.xml"),
-                        std::make_tuple("shape_of.xml")));
+        testing::Values(std::make_tuple("add_abc.xml", 0, nullptr),
+                        std::make_tuple("add_abc_f64.xml", 0, nullptr),
+                        std::make_tuple("split_equal_parts_2d.xml", 0, nullptr),
+                        std::make_tuple("addmul_abc.xml", 0, nullptr),
+                        std::make_tuple("add_abc_initializers.xml", 0, nullptr),
+                        std::make_tuple("experimental_detectron_roi_feature_extractor.xml", 0, nullptr),
+                        std::make_tuple("experimental_detectron_detection_output.xml", 0, nullptr),
+                        std::make_tuple("nms5.xml", 0, nullptr),
+                        std::make_tuple("ti_negative_stride.xml", 3149864, +[](InferenceEngine::Blob::Ptr& weights) {
+                                                        auto *data = weights->buffer().as<int64_t *>();
+                                                        data[0] = 1;
+                                                        data[1] = 512;
+                                                        data[393730] = 1;
+                                                        data[393731] = 1;
+                                                        data[393732] = 256;}),
+                        std::make_tuple("ti_resnet.xml", 8396840, +[](InferenceEngine::Blob::Ptr& weights) {
```

Comment:
@sdurawa I think it's better to move TensorIterator tests to separate file and revert tests to previous state*. For now it seems that dynamically generated weights are needed for TI & Loop only, so this is a special case and it's better to not complicate test logic for general case. 

** I've noticed an issue in previous code. CNNNetwork is constructed from xml file only, but should also take bin file into account. Please try to fix this. There are .bin files for most of the models tested here (e.g. add_abc.bin) if it is not present in repo then use empty string.  Something like this:
 ```
testing::Values(std::make_tuple("add_abc.xml","add_abc.bin"),  std::make_tuple("nms5.xml", ""))
```

Reply:
@jdanieck, current serialization test class takes into account binary file without specific inclusion in make_tuple(). In setUp() method: 
`m_out_xml_path = test_name + ".xml";`
`m_out_bin_path = test_name + ".bin";`
When performing a test following code is used:
`expected = ie.ReadNetwork(m_model_path);`
It reads bin file with the same name as xml and if bin file with the same name was not found, will load IR without weights.

Reply:
Ok then, thx Szymon. Didn't notice that ReadNetwork will create this path internally. So knowing that I'd say that calling ReadNetwork with bin path explicitly is not mandatory, but either way it is preferred in my mind because it shows intent better and you don't need to know API details to understand behaviour on the caller side. Your call, I won't insist on this change.

Reply:
New approach is implemented.